src/NServiceBus.RabbitMQ/Configuration/ConnectionConfiguration.cs

```diff
@@ -2,58 +2,145 @@
+
+            var invalidOptionsMessage = new StringBuilder();
+
+            if (dictionary.ContainsKey("dequeuetimeout"))
+            {
+                invalidOptionsMessage.AppendLine("The 'DequeueTimeout' connection string option has been removed. Consult the documentation for further information.");
+            }
+
+            if (dictionary.ContainsKey("maxwaittimeforconfirms"))
+            {
+                invalidOptionsMessage.AppendLine("The 'MaxWaitTimeForConfirms' connection string option has been removed. Consult the documentation for further information");
+            }
+
+            if (dictionary.ContainsKey("prefetchcount"))
+            {
+                invalidOptionsMessage.AppendLine("The 'PrefetchCount' connection string option has been removed. Use 'EndpointConfiguration.UseTransport<RabbitMQTransport>().PrefetchCount' instead.");
+            }
+
+            if (dictionary.ContainsKey("usepublisherconfirms"))
+            {
+                invalidOptionsMessage.AppendLine("The 'UsePublisherConfirms' connection string option has been removed. Use 'EndpointConfiguration.UseTransport<RabbitMQTransport>().UsePublisherConfirms' instead.");
+            }
+
+            var useTls = GetValue(dictionary, "useTls", bool.TryParse, defaultUseTls, invalidOptionsMessage);
+            var port = GetValue(dictionary, "port", int.TryParse, useTls ? defaultTlsPort : defaultPort, invalidOptionsMessage);
+            var virtualHost = GetValue(dictionary, "virtualHost", "/");
+            var userName = GetValue(dictionary, "userName", "guest");
+            var password = GetValue(dictionary, "password", "guest");
+            var requestedHeartbeat = GetValue(dictionary, "requestedHeartbeat", ushort.TryParse, defaultRequestedHeartbeat, invalidOptionsMessage);
+            var retryDelay = GetValue(dictionary, "retryDelay", TimeSpan.TryParse, defaultRetryDelay, invalidOptionsMessage);
+            var certPath = GetValue(dictionary, "certPath", defaultCertPath);
+            var certPassPhrase = GetValue(dictionary, "certPassphrase", defaultCertPassphrase);
+
+            var host = default(string);
+            string value;
+            if (dictionary.TryGetValue("host", out value))
+            {
+                var hostsAndPorts = value.Split(',');
+
+                if (hostsAndPorts.Length > 1)
+                {
+                    invalidOptionsMessage.AppendLine("Multiple hosts are no longer supported. If using RabbitMQ in a cluster, consider using a load balancer to represent the nodes as a single host.");
+                }
+
+                var parts = hostsAndPorts[0].Split(':');
+                host = parts.ElementAt(0);
+                if (host.Length == 0)
+                {
+                    invalidOptionsMessage.AppendLine("Empty host name in 'host' connection string option.");
+                }
+
+                if (parts.Length > 1 && !int.TryParse(parts[1], out port))
```

Comment:
Spent more time looking at this, and this looks fine. In fact, it looks like it handles even more parsing scenarios than the previous code did, so 👍.